Python/Product/PythonTools/PythonTools/LanguageServerClient/FileWatcher/Listener.cs

```diff
@@ -0,0 +1,176 @@
+﻿using System;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+using Microsoft.Extensions.FileSystemGlobbing;
+using Microsoft.PythonTools.Interpreter;
+using Microsoft.VisualStudio.LanguageServer.Protocol;
+using Microsoft.VisualStudio.Workspace;
+using Microsoft.VisualStudio.Workspace.VSIntegration.Contracts;
+using StreamJsonRpc;
+
+namespace Microsoft.PythonTools.LanguageServerClient.FileWatcher {
+    class Listener : IDisposable {
+        private readonly JsonRpc _rpc;
+        private System.IO.FileSystemWatcher _solutionWatcher;
+        private Microsoft.Extensions.FileSystemGlobbing.Matcher _matcher = new Microsoft.Extensions.FileSystemGlobbing.Matcher(StringComparison.InvariantCultureIgnoreCase);
```

Comment:
This object here can use globs to match file names. It's how I'm doing the glob lookup for the files to watch from pylance.